### PR TITLE
(PC-7024) App native - Favoris - Affichage sans favoris (état connecté)

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-7024
## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [S9] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |  image  |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/favorites/components/__tests__/FavoritesResults.test.tsx [hunk 1]
```diff
@@ -13,29 +17,32 @@ jest.mock('features/favorites/pages/FavoritesWrapper', () => ({
   }),
 }))
 
+jest.mock('features/favorites/pages/useFavorites')
+const mockUseFavorites = useFavorites as jest.Mock
```
Comment (BenStirrup) [inline src/features/favorites/components/__tests__/FavoritesResults.test.tsx:21]: Je vais faire comme si je n'avais pas vu 🙈  :
```suggestion
const mockUseFavorites = useFavorites as jest.MockedFunction<...>
```

File: src/features/favorites/components/__tests__/FavoritesResults.test.tsx [hunk 2]
```diff
@@ -13,29 +17,32 @@ jest.mock('features/favorites/pages/FavoritesWrapper', () => ({
...
-  useFavoritesResults: () => ({
-    data: mockData,
-    isFetching: false,
-    hasNextPage: mockHasNextPage,
-    fetchNextPage: mockFetchNextPage,
-  }),
-}))
+
+jest.mock('features/favorites/pages/useFavoritesResults')
+const mockUseFavoritesResults = useFavoritesResults as jest.Mock
```
Comment (BenStirrup) [inline src/features/favorites/components/__tests__/FavoritesResults.test.tsx:28]:
```suggestion
const mockUseFavoritesResults = useFavoritesResults as jest.MockedFunction<...>
```

**Review by BenStirrup — COMMENTED** [review]
Jolis tests, juste une remarque sur le typage

File: src/features/favorites/components/__tests__/FavoritesResults.test.tsx (hunk 1 above)
Comment (kopax-polyconseil) [reply src/features/favorites/components/__tests__/FavoritesResults.test.tsx:21]: Oui pardon faut s'y habituer, c'est OK

File: src/features/favorites/components/__tests__/FavoritesResults.test.tsx (hunk 2 above)
Comment (kopax-polyconseil) [reply src/features/favorites/components/__tests__/FavoritesResults.test.tsx:28]: OK

**Pull request merged**